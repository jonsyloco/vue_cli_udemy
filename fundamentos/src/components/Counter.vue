<template>
  <h2>Counter</h2>
  <p>{{ counter }} <sup>2</sup> = {{ getSquareCounter }}</p>
  <p>{{ counter }} <sup>2</sup> = {{ getSquareCounter }}</p>
  <p>{{ counter }} <sup>2</sup> = {{ getSquareCounter }}</p>
  <p>{{ counter }} <sup>2</sup> = {{ getSquareCounter }}</p>
  <p>{{ counter }} <sup>2</sup> = {{ getSquareCounter }}</p>
  <p>{{ counter }} <sup>2</sup> = {{ getSquareCounter }}</p>
 
    <div>
        <button>+1</button>
        <button>-1</button>
    </div>
</template>

<script>
export default {
  name: "Counter",
  data() {
    return {
      counter: 5,
    };
  },
  methods: {
    getSquareValue() {
        console.log("getSquareValue");
        return this.counter * this.counter;
    },
  },
  computed: {
      getSquareCounter() {
          console.log("computed");
          return this.getSquareValue();
      }
  },
};
</script>

<style>
    button{
        background-color: aquamarine;
        border-radius: 5px;
        border: 1px solid white;
        cursor: pointer;
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    button:hover{
        background-color: yellow;
        transition: 0.3s;
    }
</style>