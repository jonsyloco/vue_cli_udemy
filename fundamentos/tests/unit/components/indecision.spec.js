import { shallowMount } from '@vue/test-utils';
import  Counter  from '@/components/Counter';

describe('Indecision Component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Counter)                
    });
    test('match con el snapshot', () =>{
        expect(wrapper.html() ).toMatchSnapshot()
    })
});