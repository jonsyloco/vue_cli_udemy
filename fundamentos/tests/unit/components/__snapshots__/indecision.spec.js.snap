// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision Component match con el snapshot 1`] = `
<div>
  <h1>Counter</h1>
  <p>5 <sup>2</sup> = 25</p>
  <p data-testid="counter">5</p>
  <div class="botones"><button>+1</button><button>-1</button></div>
</div>
`;
